python/lsst/ip/isr/isrTaskLSST.py

```diff
@@ -1835,16 +1841,37 @@ def run(self, ccdExposure, *, dnlLUT=None, bias=None, deferredChargeCalib=None,
                 maskPlaneNames=self.config.itlDipMaskPlanes,
             )
 
-        # Edge bleed masking
-        if self.config.doITLEdgeBleedMask and detector.getPhysicalType() == 'ITL' \
-                and self.config.doSaturation:
-            isrFunctions.maskITLEdgeBleed(ccdExposure=ccdExposure,
-                                          itlEdgeBleedSatMinArea=self.config.itlEdgeBleedSatMinArea,
-                                          itlEdgeBleedSatMaxArea=self.config.itlEdgeBleedSatMaxArea,
-                                          itlEdgeBleedThreshold=self.config.itlEdgeBleedThreshold,
-                                          itlEdgeBleedModelConstant=self.config.itlEdgeBleedModelConstant,
-                                          saturatedMaskName=self.config.saturatedMaskName,
-                                          badAmpDict=badAmpDict)
+        if (self.config.doITLSatSagMask or self.config.doITLEdgeBleedMask) \
+                and detector.getPhysicalType() == 'ITL':
+
+            # The following steps will rely on the footprint of saturated
+            # cores with large areas.
+            thresh = afwDetection.Threshold(ccdExposure.mask.getPlaneBitMask("SAT"),
+                                            afwDetection.Threshold.BITMASK
+                                            )
+            fpList = afwDetection.FootprintSet(ccdExposure.mask, thresh).getFootprints()
+
+            satAreas = numpy.asarray([fp.getArea() for fp in fpList])
+            largeAreas, = numpy.where((satAreas >= self.config.itlEdgeBleedSatMinArea)
+                                      & (satAreas < self.config.itlEdgeBleedSatMaxArea))
+
+            for largeAreasIndex in largeAreas:
+
+                fpCore = fpList[largeAreasIndex]
+
+                # Edge bleed masking
+                if self.config.doITLEdgeBleedMask:
+                    isrFunctions.maskITLEdgeBleed(ccdExposure=ccdExposure,
```

Comment:
I think I'd suggest just putting `config = self.config` in this block, and using the 5 character shorter version to get around the lint.

Reply:
If this goes into a method it'll be reduced at least one indentation level which will also help.